activerecord/lib/active_record/relation/delegation.rb

```diff
@@ -5,19 +5,31 @@
 
 module ActiveRecord
   module Delegation # :nodoc:
+    class << self
+      def delegated_classes
+        [
+          ActiveRecord::Relation,
+          ActiveRecord::Associations::CollectionProxy,
+          ActiveRecord::AssociationRelation,
+          ActiveRecord::DisableJoinsAssociationRelation,
+        ]
+      end
+
+      def uncacheable_methods
+        @uncacheable_methods ||= (
```

Comment:
I wish I could make this a constant instead, but it cause load order issues. Which is yet another argument to just remove that caching.